### PR TITLE
Models section

### PR DESCRIPTION
**Pull request by mkju19**

### CODE REVIEW

File: sections/Chapter 5 Sprint 3/Models.tex [hunk 1]
```diff
@@ -0,0 +1,65 @@
+\subsection{Models}
+To implement an API for the WordCount database we needed models for the objects we would be sending and recieving from the other layers.
```
Comment (Pattrigue) [inline sections/Chapter 5 Sprint 3/Models.tex:2]:
```suggestion
To implement an API for the WordCount database, we needed models for the objects we would be sending and receiving from the other layers. 
```

File: sections/Chapter 5 Sprint 3/Models.tex [hunk 2]
```diff
@@ -0,0 +1,65 @@
+\subsection{Models}
+To implement an API for the WordCount database we needed models for the objects we would be sending and recieving from the other layers. 
+We also needed models corresponding to the database tables to access it through EF Core. The models we have created for communication with other layers and for data access will be described in the following sections.
```
Comment (Pattrigue) [inline sections/Chapter 5 Sprint 3/Models.tex:3]: It's not clear if these two models are the same

File: sections/Chapter 5 Sprint 3/Models.tex [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+
+\subsubsection*{Communication models}
+For communication between our layer and the other layers, we have made 3 input, and 1 response model.
```
Comment (Pattrigue) [inline sections/Chapter 5 Sprint 3/Models.tex:6]:
```suggestion
For communication between our layer and the other layers, we have made three input models and one response model. 
```

File: sections/Chapter 5 Sprint 3/Models.tex [hunk 4]
```diff
@@ -0,0 +1,65 @@
+\subsection{Models}
+To implement an API for the WordCount database we needed models for the objects we would be sending and recieving from the other layers. 
+We also needed models corresponding to the database tables to access it through EF Core. The models we have created for communication with other layers and for data access will be described in the following sections.
+
+\subsubsection*{Communication models}
+For communication between our layer and the other layers, we have made 3 input, and 1 response model. 
+These models correspond to the JSON objects we expect to send or recieve from other layers when they attempt to insert or retrive data from the database.
```
Comment (Pattrigue) [inline sections/Chapter 5 Sprint 3/Models.tex:7]:
```suggestion
These models correspond to the JSON objects we expect to send or receive from other layers when they attempt to insert data into or retrieve data from the database. 
```

File: sections/Chapter 5 Sprint 3/Models.tex [hunk 5]
```diff
@@ -0,0 +1,65 @@
+\subsection{Models}
+To implement an API for the WordCount database we needed models for the objects we would be sending and recieving from the other layers. 
+We also needed models corresponding to the database tables to access it through EF Core. The models we have created for communication with other layers and for data access will be described in the following sections.
+
+\subsubsection*{Communication models}
+For communication between our layer and the other layers, we have made 3 input, and 1 response model. 
+These models correspond to the JSON objects we expect to send or recieve from other layers when they attempt to insert or retrive data from the database. 
+When a JSON object is recieved it is parsed into an object from our model, which can then be used to add data to the database.
```
Comment (Pattrigue) [inline sections/Chapter 5 Sprint 3/Models.tex:8]:
```suggestion
When a JSON object is received, it is parsed into an object from our model which can then be used to add data to the database.
```

File: sections/Chapter 5 Sprint 3/Models.tex [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+\subsubsection*{Communication models}
+For communication between our layer and the other layers, we have made three input models and one response model. 
+These models correspond to the JSON objects we expect to send or receive from other layers when they attempt to insert data into or retrieve data from the database. 
+When a JSON object is received, it is parsed into an object from our model, which can then be used to add data to the database.
+\\
+
+\textbf{Input models}\\
+% ArticleJsonModel
+% TermJsonModel
+The first input model is the ArticleJsonModel which can be seen in figure \ref{ArticlJsonModel}. An article consists of a title, a publisher, a file path, the number of words in an article, and an array containing all the words in the article.
```
Comment (Pattrigue) [inline sections/Chapter 5 Sprint 3/Models.tex:14]:
```suggestion
The first input model is \texttt{ArticleJsonModel} which can be seen in figure \ref{ArticlJsonModel}. An article consists of a title, a publisher, a file path, the number of words in the article and an array containing all the words in the article. 
```

File: sections/Chapter 5 Sprint 3/Models.tex [hunk 7]
```diff
@@ -0,0 +1,65 @@
...
+For communication between our layer and the other layers, we have made three input models and one response model. 
+These models correspond to the JSON objects we expect to send or receive from other layers when they attempt to insert data into or retrieve data from the database. 
+When a JSON object is received, it is parsed into an object from our model, which can then be used to add data to the database.
+\\
+
+\textbf{Input models}\\
+% ArticleJsonModel
+% TermJsonModel
+The first input model is the ArticleJsonModel which can be seen in figure \ref{ArticlJsonModel}. An article consists of a title, a publisher, a file path, the number of words in an article, and an array containing all the words in the article. 
+The words are of the type \texttt{TermJsonModel} which has the word itself and how many times it occurs in the article.
```
Comment (Pattrigue) [inline sections/Chapter 5 Sprint 3/Models.tex:15]:
```suggestion
The words are of type \texttt{TermJsonModel} which is a separate model containing the word itself and a number denoting how many times it occurs in the given article.
```

File: sections/Chapter 5 Sprint 3/Models.tex [hunk 8]
```diff
@@ -0,0 +1,65 @@
...
+This model is used when the database receives a JSON object through our endpoint, which corresponds to the ArticleJsonModel.
```
Comment (Pattrigue) [inline sections/Chapter 5 Sprint 3/Models.tex:16]:
```suggestion
This model is used when the database receives a JSON object through our endpoint, which corresponds to the \texttt{ArticleJsonModel}. 
```

File: sections/Chapter 5 Sprint 3/Models.tex [hunk 9]
```diff
@@ -0,0 +1,65 @@
...
+When a JSON object is received, it is parsed into an object from our model, which can then be used to add data to the database.
+\\
+
+\textbf{Input models}\\
+% ArticleJsonModel
+% TermJsonModel
+The first input model is the ArticleJsonModel which can be seen in figure \ref{ArticlJsonModel}. An article consists of a title, a publisher, a file path, the number of words in an article, and an array containing all the words in the article. 
+The words are of the type \texttt{TermJsonModel} which has the word itself and how many times it occurs in the article.
+This model is used when the database receives a JSON object through our endpoint, which corresponds to the ArticleJsonModel. 
+After the JSON object has been converted to a \texttt{ArticleJsonModel}, the data from the article can be inserted into the \texttt{WordCount} database using EF Core.
```
Comment (Pattrigue) [inline sections/Chapter 5 Sprint 3/Models.tex:17]:
```suggestion
After the JSON object has been converted to \texttt{ArticleJsonModel}, the data from the article can be inserted into the WordCount database using EF Core.
```

File: sections/Chapter 5 Sprint 3/Models.tex [hunk 10]
```diff
@@ -0,0 +1,65 @@
...
+
+\begin{figure}[H]
+    \centering
+    \includegraphics[scale=0.5]{Images/jsonArticleModel.PNG}
+    \caption{Diagram of the \texttt{ArticlJsonModel} and the \texttt{TermJsonModel}}
+    \label{ArticlJsonModel}
+\end{figure}
+
+% JsonSchemaInputModel
+The second input model is the \texttt{JsonSchemaInputModel} which can be seen in figure \ref*{JsonSchemaInputModel}. The \texttt{JsonSchemaInputModel} contains a name of the JSON schema, and a body which is the JSON schema itself.
```
Comment (Pattrigue) [inline sections/Chapter 5 Sprint 3/Models.tex:27]:
```suggestion
The second input model is \texttt{JsonSchemaInputModel} which can be seen in figure \ref*{JsonSchemaInputModel}. \texttt{JsonSchemaInputModel} contains a name of the JSON schema, and a body which is the JSON schema itself. 
```

File: sections/Chapter 5 Sprint 3/Models.tex [hunk 11]
```diff
@@ -0,0 +1,65 @@
...
+\begin{figure}[H]
+    \centering
+    \includegraphics[scale=0.5]{Images/jsonArticleModel.PNG}
+    \caption{Diagram of the \texttt{ArticlJsonModel} and the \texttt{TermJsonModel}}
+    \label{ArticlJsonModel}
+\end{figure}
+
+% JsonSchemaInputModel
+The second input model is the \texttt{JsonSchemaInputModel} which can be seen in figure \ref*{JsonSchemaInputModel}. The \texttt{JsonSchemaInputModel} contains a name of the JSON schema, and a body which is the JSON schema itself. 
+The \texttt{JsonSchemaInputModel} is used when the other layers try to add JSON schemas to the \texttt{WordCount} database, which is then later used to validate their future input.
```
Comment (Pattrigue) [inline sections/Chapter 5 Sprint 3/Models.tex:28]:
```suggestion
It is used when the other layers try to add JSON schemas to the WordCount database, which is then later used to validate future input. 
```

File: sections/Chapter 5 Sprint 3/Models.tex [hunk 12]
```diff
@@ -0,0 +1,65 @@
...
+
+% JsonSchemaInputModel
+The second input model is the \texttt{JsonSchemaInputModel} which can be seen in figure \ref*{JsonSchemaInputModel}. The \texttt{JsonSchemaInputModel} contains a name of the JSON schema, and a body which is the JSON schema itself. 
+The \texttt{JsonSchemaInputModel} is used when the other layers try to add JSON schemas to the \texttt{WordCount} database, which is then later used to validate their future input. 
+How the validation is done will be described in section \ref*{}. \todo{insert ref to validation section}.
+
+\begin{figure}[H]
+    \centering
+    \includegraphics[scale=0.25]{Images/JsonSchemaInputModel.png}
+    \caption{Diagram of the \texttt{JsonSchemaInputModel}}
```
Comment (Pattrigue) [inline sections/Chapter 5 Sprint 3/Models.tex:34]:
```suggestion
    \caption{Diagram of the \texttt{JsonSchemaInputModel}.}
```

File: sections/Chapter 5 Sprint 3/Models.tex [hunk 13]
```diff
@@ -0,0 +1,65 @@
...
+% TermJsonModel
+The first input model is the ArticleJsonModel which can be seen in figure \ref{ArticlJsonModel}. An article consists of a title, a publisher, a file path, the number of words in an article, and an array containing all the words in the article. 
+The words are of the type \texttt{TermJsonModel} which has the word itself and how many times it occurs in the article.
+This model is used when the database receives a JSON object through our endpoint, which corresponds to the ArticleJsonModel. 
+After the JSON object has been converted to a \texttt{ArticleJsonModel}, the data from the article can be inserted into the \texttt{WordCount} database using EF Core.
+
+\begin{figure}[H]
+    \centering
+    \includegraphics[scale=0.5]{Images/jsonArticleModel.PNG}
+    \caption{Diagram of the \texttt{ArticlJsonModel} and the \texttt{TermJsonModel}}
```
Comment (Pattrigue) [inline sections/Chapter 5 Sprint 3/Models.tex:22]:
```suggestion
    \caption{Diagram of the \texttt{ArticlJsonModel} and the \texttt{TermJsonModel}.}
```

File: sections/Chapter 5 Sprint 3/Models.tex [hunk 14]
```diff
@@ -0,0 +1,65 @@
...
+\begin{figure}[H]
+    \centering
+    \includegraphics[scale=0.25]{Images/JsonSchemaInputModel.png}
+    \caption{Diagram of the \texttt{JsonSchemaInputModel}}
+    \label{JsonSchemaInputModel}
+\end{figure}
+
+\textbf{Response Models}\\
+% FileIdResponse
+The response model \texttt{FileIdResponse} which can be seen in figure \ref*{FileIdResponse}is used when a request for the location of an article is made.
```
Comment (Pattrigue) [inline sections/Chapter 5 Sprint 3/Models.tex:40]:
```suggestion
The response model \texttt{FileIdResponse} is used when a request for the path of an article is made, and can be seen in figure \ref*{FileIdResponse},
```

File: sections/Chapter 5 Sprint 3/Models.tex [hunk 15]
```diff
@@ -0,0 +1,65 @@
...
+    \centering
+    \includegraphics[scale=0.25]{Images/JsonSchemaInputModel.png}
+    \caption{Diagram of the \texttt{JsonSchemaInputModel}}
+    \label{JsonSchemaInputModel}
+\end{figure}
+
+\textbf{Response Models}\\
+% FileIdResponse
+The response model \texttt{FileIdResponse} which can be seen in figure \ref*{FileIdResponse}is used when a request for the location of an article is made. 
+After the location has been queried on the \texttt{WordCount} database, it is put into the response model, which is returned as a JSON object.
```
Comment (Pattrigue) [inline sections/Chapter 5 Sprint 3/Models.tex:41]:
```suggestion
After the path has been queried in the WordCount database, it is put into the response model which is returned as a JSON object.
```

File: sections/Chapter 5 Sprint 3/Models.tex [hunk 16]
```diff
@@ -0,0 +1,65 @@
...
+
+\begin{figure}[H]
+    \centering
+    \includegraphics[scale=0.25]{Images/FileIdResponse.png}
+    \caption{Diagram of the \texttt{FileIdResponse} model}
```
Comment (Pattrigue) [inline sections/Chapter 5 Sprint 3/Models.tex:46]:
```suggestion
    \caption{Diagram of the \texttt{FileIdResponse} model.}
```

File: sections/Chapter 5 Sprint 3/Models.tex [hunk 17]
```diff
@@ -0,0 +1,65 @@
...
+    \label{FileIdResponse}
+\end{figure}
+
+\subsubsection*{Data access models}
+To use EF Core for data access, we also had to implement models of the tables in the database, which can then be used to insert and query data on the database.
```
Comment (Pattrigue) [inline sections/Chapter 5 Sprint 3/Models.tex:51]:
```suggestion
To use EF Core for data access, we also had to implement models of the relations in the database, which are be used to insert and query data. 
```

File: sections/Chapter 5 Sprint 3/Models.tex [hunk 18]
```diff
@@ -0,0 +1,65 @@
...
+\begin{figure}[H]
+    \centering
+    \includegraphics[scale=0.25]{Images/FileIdResponse.png}
+    \caption{Diagram of the \texttt{FileIdResponse} model}
+    \label{FileIdResponse}
+\end{figure}
+
+\subsubsection*{Data access models}
+To use EF Core for data access, we also had to implement models of the tables in the database, which can then be used to insert and query data on the database. 
+All of these models must be identical to what is on the database, and they are all used the same way by EF Core to manipulate data.
```
Comment (Pattrigue) [inline sections/Chapter 5 Sprint 3/Models.tex:52]:
```suggestion
All of these models must be contain exactly the same fields as the relations in the database.
EF Core then maps these models to the relations such that they can be used to manipulate data.
```

File: sections/Chapter 5 Sprint 3/Models.tex [hunk 19]
```diff
@@ -0,0 +1,65 @@
...
+\\
+% Article
+% publisher
+% Term
+The model of the \texttt{Article} table along with a \texttt{Term} and a \texttt{Publisher} can be seen in figure \ref*{Article}.
```
Comment (Pattrigue) [inline sections/Chapter 5 Sprint 3/Models.tex:57]:
```suggestion
The model of the \texttt{Article} relation as well as \texttt{Term} and \texttt{Publisher} can be seen in figure \ref*{Article}. 
```

File: sections/Chapter 5 Sprint 3/Models.tex [hunk 20]
```diff
@@ -0,0 +1,65 @@
...
+All of these models must be identical to what is on the database, and they are all used the same way by EF Core to manipulate data.
+\\
+% Article
+% publisher
+% Term
+The model of the \texttt{Article} table along with a \texttt{Term} and a \texttt{Publisher} can be seen in figure \ref*{Article}. 
+\begin{figure}[H]
+    \centering
+    \includegraphics[scale=0.25]{Images/ArticleModel.PNG}
+    \caption{Article}
```
Comment (Pattrigue) [inline sections/Chapter 5 Sprint 3/Models.tex:61]: Need a better caption than that.

File: sections/Chapter 5 Sprint 3/Models.tex [hunk 21]
```diff
@@ -0,0 +1,65 @@
...
+    \label{Article}
+\end{figure}
+The other models used for database access have  similarly been created to replicate the structure of the database.
```
Comment (Pattrigue) [inline sections/Chapter 5 Sprint 3/Models.tex:64]:
```suggestion
Similarly, the other models used for database access have been created to replicate the structure of the database. 
```

File: sections/Chapter 5 Sprint 3/Models.tex [hunk 22]
```diff
@@ -0,0 +1,65 @@
...
+% Term
+The model of the \texttt{Article} table along with a \texttt{Term} and a \texttt{Publisher} can be seen in figure \ref*{Article}. 
+\begin{figure}[H]
+    \centering
+    \includegraphics[scale=0.25]{Images/ArticleModel.PNG}
+    \caption{Article}
+    \label{Article}
+\end{figure}
+The other models used for database access have  similarly been created to replicate the structure of the database. 
+The other models can be seen in appendix \ref*{AppDataAccess}.
```
Comment (Pattrigue) [inline sections/Chapter 5 Sprint 3/Models.tex:65]:
```suggestion
The remaining models can be seen in appendix \ref*{AppDataAccess}.
```

File: sections/appAname.tex [hunk 23]
```diff
@@ -1,2 +1,18 @@
 \chapter{Appendix A name}\label{ch:appAlabel}
 Here is the first appendix
+
+\chapter{Data access models}\label{AppDataAccess}
+% % JsonSchemaModel
+\begin{figure}[H]
+    \centering
+    \includegraphics[scale=0.25]{Images/JsonSchemaModel.png}
+    \caption{The model of the \texttt{JsonSchema} table}
```
Comment (Pattrigue) [inline sections/appAname.tex:9]:
```suggestion
    \caption{The model of the \texttt{JsonSchema} table.}
```

File: sections/appAname.tex [hunk 24]
```diff
@@ -1,2 +1,18 @@
...
+    \label{JsonSchemaModel}
+\end{figure}
+% WordRatio
+\begin{figure}[H]
+    \centering
+    \includegraphics[scale=0.25]{Images/WordRatioModel.png}
+    \caption{WordRatioModel}
```
Comment (Pattrigue) [inline sections/appAname.tex:16]: Need a better caption than that

File: sections/Chapter 5 Sprint 3/Models.tex [hunk 25]
```diff
@@ -0,0 +1,65 @@
+\subsection{Models}
+To implement an API for the WordCount database, we needed models for the objects we would be sending and receiving from the other layers. 
+We also required models corresponding to the database tables to access it through EF Core. The models we have created for communication with other layers and for data access will be described in the following sections.
```
Comment (Pattrigue) [inline sections/Chapter 5 Sprint 3/Models.tex:3]: Sentence is ambiguous

File: sections/Chapter 5 Sprint 3/Models.tex (hunk 20 above)
Comment (mkju19) [reply sections/Chapter 5 Sprint 3/Models.tex:62]:
```suggestion
    \caption{Diagram of the Article model}
```

File: sections/appAname.tex (hunk 24 above)
Comment (mkju19) [reply sections/appAname.tex:16]:
```suggestion
    \caption{Diagram of WordRatioModel}
```

File: sections/Chapter 5 Sprint 3/Models.tex (hunk 25 above)
Comment (mkju19) [reply sections/Chapter 5 Sprint 3/Models.tex:3]:
```suggestion
We also required seperate models corresponding to the database tables to access it through EF Core. The models we have created for communication with other layers and for data access will be described in the following sections.
```

File: sections/appAname.tex [hunk 26]
```diff
@@ -1,2 +1,18 @@
...
+    \centering
+    \includegraphics[scale=0.25]{Images/JsonSchemaModel.png}
+    \caption{The model of the \texttt{JsonSchema} table.}
+    \label{JsonSchemaModel}
+\end{figure}
+% WordRatio
+\begin{figure}[H]
+    \centering
+    \includegraphics[scale=0.25]{Images/WordRatioModel.png}
+    \caption{Diagram of WordRatioModel}
```
Comment (Pattrigue) [inline sections/appAname.tex:16]:
```suggestion
    \caption{Diagram of WordRatioModel.}
```

**Review by Pattrigue — APPROVED** [review]

**Pull request merged**